metrics-core/src/main/java/io/dropwizard/metrics/MetricRegistry.java

```diff
@@ -271,8 +271,13 @@ public void removeListener(MetricRegistryListener listener) {
      * @param filter    the metric filter to match
      * @return all the gauges in the registry
      */
-    public SortedMap<MetricName, Gauge> getGauges(MetricFilter filter) {
-        return getMetrics(Gauge.class, filter);
+    public SortedMap<MetricName, Gauge<?>> getGauges(MetricFilter filter) {
+        final SortedMap<MetricName, Gauge> metrics = getMetrics(Gauge.class, filter);
+        final SortedMap<MetricName, Gauge<?>> result = new TreeMap<>();
```

Comment:
Could we avoid creating a copy of `metrics`? I guess `javac` doesn't allow to cast `SortedMap<MetricName, Gauge>` to `SortedMap<MetricName, Gauge<?>>`, but I would like to avoid creating an additional copy just because of compiler peculiarities.